File: CS/Q215883/App_Code/PersistentClasses.cs
using System;
using DevExpress.Xpo;

public class Customer : XPObject {
    public Customer(Session session) : base(session) { }

    private string fName;
    public string Name {
        get { return fName; }
        set { SetPropertyValue<string>("Name", ref fName, value); }
    }

    private int fAge;
    public int Age {
        get { return fAge; }
        set { SetPropertyValue<int>("Age", ref fAge, value); }
    }

    [Association("Customer-Orders", typeof(Order))]
    public XPCollection<Order> Orders { get { return GetCollection<Order>("Orders"); } }
}

public class Order : XPObject {
    public Order(Session session) : base(session) { }

    private string fOrderName;
    public string OrderName {
        get { return fOrderName; }
        set { SetPropertyValue<string>("OrderName", ref fOrderName, value); }
    }

    private DateTime fOrderDate;
    public DateTime OrderDate {
        get { return fOrderDate; }
        set { SetPropertyValue<DateTime>("OrderDate", ref fOrderDate, value); }
    }

    private Customer fCustomer;
    [Association("Customer-Orders")]
    public Customer Customer {
        get { return fCustomer; }
        set { SetPropertyValue<Customer>("Customer", ref fCustomer, value); }
    }
}

File: VB/Q215883/App_Code/PersistentClasses.vb
Imports System
Imports DevExpress.Xpo

Public Class Customer
	Inherits XPObject

	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	Private fName As String
	Public Property Name() As String
		Get
			Return fName
		End Get
		Set(ByVal value As String)
			SetPropertyValue(Of String)("Name", fName, value)
		End Set
	End Property

	Private fAge As Integer
	Public Property Age() As Integer
		Get
			Return fAge
		End Get
		Set(ByVal value As Integer)
			SetPropertyValue(Of Integer)("Age", fAge, value)
		End Set
	End Property

	<Association("Customer-Orders", GetType(Order))>
	Public ReadOnly Property Orders() As XPCollection(Of Order)
		Get
			Return GetCollection(Of Order)("Orders")
		End Get
	End Property
End Class

Public Class Order
	Inherits XPObject

	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	Private fOrderName As String
	Public Property OrderName() As String
		Get
			Return fOrderName
		End Get
		Set(ByVal value As String)
			SetPropertyValue(Of String)("OrderName", fOrderName, value)
		End Set
	End Property

	Private fOrderDate As DateTime
	Public Property OrderDate() As DateTime
		Get
			Return fOrderDate
		End Get
		Set(ByVal value As DateTime)
			SetPropertyValue(Of DateTime)("OrderDate", fOrderDate, value)
		End Set
	End Property

	Private fCustomer As Customer
	<Association("Customer-Orders")>
	Public Property Customer() As Customer
		Get
			Return fCustomer
		End Get
		Set(ByVal value As Customer)
			SetPropertyValue(Of Customer)("Customer", fCustomer, value)
		End Set
	End Property
End Class